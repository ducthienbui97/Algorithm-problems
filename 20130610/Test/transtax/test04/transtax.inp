175 175
93 152
115 175
16 33
127 156
70 142
28 153
79 127
102 116
68 77
81 156
10 97
46 63
65 131
12 24
137 161
44 104
27 49
38 110
93 173
46 113
25 116
61 154
33 174
117 130
96 157
86 148
114 134
29 62
111 144
43 69
23 149
59 78
4 110
8 171
98 143
15 65
32 126
88 142
56 152
4 135
51 91
116 135
117 120
83 143
75 164
10 147
57 108
65 172
29 110
63 123
63 71
144 165
31 144
54 128
13 136
2 6
48 117
24 127
2 133
17 158
33 111
129 135
25 26
40 118
4 156
2 39
85 87
5 152
87 103
71 142
100 127
23 115
96 109
23 51
47 60
72 97
126 163
64 146
44 121
27 115
86 152
134 159
14 135
25 58
46 151
21 102
41 155
129 147
55 93
25 76
143 170
50 139
34 118
13 98
44 84
126 127
73 154
44 96
27 139
99 173
88 150
55 57
131 168
63 105
3 171
33 112
28 52
33 160
49 122
28 129
68 137
15 90
167 169
22 137
17 73
37 86
71 116
27 154
106 146
96 141
95 129
2 135
156 170
101 129
80 156
64 142
45 164
114 162
40 93
36 77
12 67
86 119
56 145
20 173
71 131
83 85
14 32
18 107
125 171
59 74
131 171
43 73
33 169
71 74
30 102
120 130
41 88
89 123
11 171
138 161
6 82
77 92
29 140
35 110
156 166
7 110
66 133
35 124
25 47
154 164
42 115
94 139
1 14
42 48
19 142
9 133
73 137
142 174
97 107
73 128
53 80
43 134
43 132
96 115
56 116
