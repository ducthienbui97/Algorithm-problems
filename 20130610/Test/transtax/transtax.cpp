/*
PROG: cowfun
LANG: C++
pos: your_id_here
*/
using namespace std;
#include<cstdio>
#include<cstdlib>
#include<cstring>
#include<sstream>
#include<algorithm>
#include<vector>
#include<list>
#include<map>
#include<set>
#include<queue>
#include<stack>
#include<cmath>
#include<cassert>
#define FOR(i, a, b) for(int i=(a), _b=(b); i < _b; ++i)
#define FORE(i, a, b) for(int i=(a), _b=(b); i <= _b; ++i)
#define FORD(i, a, b) for(int i=(a), _b=(b); i >= _b; --i)
#define MP make_pair
#define PB push_back
#define SZ size()
#define BG begin()
#define ED end()
#define THERE(c, v) (c.find(v) != c.ED)
#define ALL(c) c.BG, c.ED
#define TR(c, it) for(typeof((c).begin()) it=(c).begin(); it!=(c).end(); it++)
#define EXIST(c, u) ( c.find(u) != c.end() )

typedef pair<int, int> II;
typedef pair<int, II> III;
typedef vector<int> VI;
typedef vector<II> VII;
typedef vector<VI> VVI;
typedef vector<VII> VVII;
typedef long long LL;

static inline int rint(){ int n; scanf("%d ", &n); return n; }
static inline void wint(const int n){ printf("%d\n", n); }
static inline void wint_(const int n){ printf("%d ", n); }
#define endl putchar('\n')
static inline void rll(LL& n){
    double t; scanf("%lf", &t); n = LL(t);
}
static inline void wll(const LL n){
    printf("%0.0lf", double(n));
}
const int DBG = 1;
const int MAXN = 10+100000;
const int MAXM = 10+200000;
const int INF = 1000000001;

char sTestFolder[1024];
char sTestInput1[1024];			// co the co nhieu input
char sTestOutput1[1024];			// co the co nhieu output
char sSubmissionOutput1[1024];	// ...

char sResult[1024];
char sDescription[1024];

void ReadComparerInput();
void ReadTestInput();
void ReadTestOutput();
void ReadSubmissionOutput();
void Report_Sai();
void Report_Dung();
void Report_SaiDinhDang();
void Report_DungMotPhan(int nPercent);

int bl[MAXN];
int m, n, ansda, ansbl;
set<II> g;

int main(int argc, char* argv[])
{
	ReadComparerInput();
	ReadTestInput();

    //sprintf(sDescription, "Doc du lieu OK"); Report_Dung();
	ReadTestOutput();
	ReadSubmissionOutput();	// Output cua thi sinh, phai kiem tra khi doc

	if(ansbl == -1){
        if(ansda == -1){
            sprintf(sDescription, "DA: -1, BL: -1");
            Report_Dung();
        }
        sprintf(sDescription, "DA: +1, BL: -1");
        Report_Sai();
    }

	FOR(i, 0, n){
        int u = i, v = bl[i]-1;
        if(u>v) swap(u, v);
        II e = II(u, v);
        if(!THERE(g, e)){
            sprintf(sDescription, "Sai o (%d, %d)", i+1, bl[i]);
            Report_Sai();
        }
        g.erase(g.find(e));
    }

	//chon mot trong 3 ham de thu nghiem
	//Report_DungMotPhan(50);
	//Report_SaiDinhDang();
	Report_Dung();
	return 0;
}

void ReadTestInput() //Du lieu
{
	FILE* f = fopen(sTestInput1, "rt");
    fscanf(f, "%d %d", &n, &m);
    int u, v;
    FOR(i, 0, m){
        fscanf(f, "%d %d", &u, &v);
        u--; v--;
        if(u>v) swap(u, v);
        g.insert(II(u, v));
    }
	fclose(f);
}

void ReadSubmissionOutput() //Bai hoc sinh
{
	FILE* f = fopen(sSubmissionOutput1, "rt");

	fscanf(f, "%d", &ansbl);

	if(ansbl == -1) return;

	FOR(i, 0, n) fscanf(f, "%d", bl+i);

	fclose(f);
}

void ReadTestOutput() //Dap an
{
	FILE* f = fopen(sTestOutput1, "rt");
	fscanf(f, "%d", &ansda);
	fclose(f);
}

void ReadComparerInput() //Dia chi file
{
	FILE* f = fopen("COMPAR$R.INP", "rt");
	fgets(sTestFolder, 255, f);
	fclose(f);

	strcpy(sTestInput1, sTestFolder);
	strcat(sTestInput1, "TRANSTAX.INP");

	strcpy(sTestOutput1, sTestFolder);
	strcat(sTestOutput1, "TRANSTAX.OUT");

	strcpy(sSubmissionOutput1, "TRANSTAX.OUT");

	strcpy(sResult, "");
	strcpy(sDescription, "");
}

void Report()
{
	FILE* f = fopen("COMPAR$R.OUT", "wt");
	fprintf(f, "%s\n%s", sResult, sDescription);
	fclose(f);
	exit(0);
}

void Report_Sai()
{
	strcpy(sResult, "SAI");
	Report();
}

void Report_Dung()
{
	strcpy(sResult, "DUNG");
	Report();
}

void Report_SaiDinhDang()
{
	strcpy(sResult, "SAIDINHDANG");
	Report();
}

void Report_DungMotPhan(int nPercent)
{
	sprintf(sResult, "DUNGMOTPHAN %d", nPercent);
	Report();
}
