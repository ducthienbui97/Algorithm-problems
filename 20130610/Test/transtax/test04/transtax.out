1
14
135
171
156
152
2
110
171
133
147
171
24
98
135
65
33
73
107
142
173
102
137
115
127
116
25
115
129
110
102
144
14
174
118
110
77
86
110
2
93
88
48
73
96
164
63
25
117
27
139
23
28
80
128
93
116
55
25
74
47
154
29
71
142
131
133
12
137
43
142
116
97
154
71
164
25
68
59
127
156
156
6
143
44
83
152
85
142
123
15
51
77
152
139
129
115
10
143
173
127
129
116
87
44
63
146
97
57
96
4
33
33
46
134
42
135
130
40
86
117
44
49
63
35
171
32
126
73
135
120
71
43
2
43
4
13
73
161
27
29
96
71
170
111
56
64
129
86
23
88
46
56
28
27
41
127
96
17
134
33
137
114
126
154
144
156
169
131
33
156
131
65
93
142
115
